compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangParserListener.java

```diff
@@ -2065,6 +2065,260 @@ public void exitStringTemplateLiteral(BallerinaParser.StringTemplateLiteralConte
+            return;
+        }
+
+        this.pkgBuilder.endWhereClauseNode(getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void enterWindowClause(BallerinaParser.WindowClauseContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.startWindowClauseNode(getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void exitWindowClause(BallerinaParser.WindowClauseContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.endWindowsClauseNode(getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void enterStreamingInput(BallerinaParser.StreamingInputContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.startStreamingInputNode(getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void exitStreamingInput(BallerinaParser.StreamingInputContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        boolean isWindowAvailable = false;
+        boolean isfirstWhereClauseAvailable = false;
+        boolean isSecondWhereClauseAvailable = false;
+        isWindowAvailable = ctx.windowClause() == null ? false : true;
+        isfirstWhereClauseAvailable = ctx.whereClause(0) == null ? false : true;
+        isSecondWhereClauseAvailable = ctx.whereClause(1) == null ? false : true;
+        String alias = null;
+        if (ctx.alias != null) {
+            alias = ctx.alias.getText();
+        }
+
+        this.pkgBuilder.endStreamingInputNode(isfirstWhereClauseAvailable, isSecondWhereClauseAvailable,
+                isWindowAvailable, alias, getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void enterJoinStreamingInput(BallerinaParser.JoinStreamingInputContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.startJoinStreamingInputNode(getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void exitJoinStreamingInput(BallerinaParser.JoinStreamingInputContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.endJoinStreamingInputNode(getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void enterTableQuery(BallerinaParser.TableQueryContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.startTableQueryNode(getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void exitTableQuery(BallerinaParser.TableQueryContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+        boolean isSelectClauseAvailable = ctx.selectClause() == null ? false : true;
+        boolean isOrderByClauseAvailable = ctx.orderByClause() == null ? false : true;
+        boolean isJoinClauseAvailable = ctx.joinStreamingInput() == null ? false : true;
```

Comment:
Simplify above 3 statements as well. No need of ternary operator here

Reply:
fixed.